client/app/src/services/network.service.js

```diff
@@ -347,22 +351,65 @@
 
     // Updates peer with all currency values relative to the USD price.
     function updatePeerWithCurrencies (peer, res) {
-      $http.get('https://api.fixer.io/latest?base=USD', {timeout: 2000}).then(function (result) {
-        const USD_PRICE = Number(res.data[0].price_usd)
-        var currencies = ['aud', 'brl', 'cad', 'chf', 'cny', 'eur', 'gbp', 'hkd', 'idr', 'inr', 'jpy', 'krw', 'mxn', 'rub']
-        var prices = {}
-        currencies.forEach(function (currency) {
-          prices[currency] = result.data.rates[currency.toUpperCase()] * USD_PRICE
-        })
-        prices['btc'] = res.data[0].price_btc
-        prices['usd'] = res.data[0].price_usd
-        peer.market.price = prices
-        storageService.setGlobal('peerCurrencies', prices)
+      peer = updateCurrencyConversionRates(peer)
+      const USD_PRICE = Number(res.data[0].price_usd)
+      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
+      var prices = {}
+      currencies.forEach(function (currency) {
+        prices[currency.toLowerCase()] = peer.market.conversionRates[currency] * USD_PRICE
       })
+      prices['btc'] = res.data[0].price_btc
+      prices['usd'] = res.data[0].price_usd
+      storageService.setGlobal('peerCurrencies', prices)
+      peer.market.price = prices
+      return peer
+    }
 
+    // Updates the currency conversion rates IF necessary
+    // Necessary if it isn't stored, or if the stored value is too old
+    function updateCurrencyConversionRates (peer) {
+      if (storageService.getGlobal('conversionRates') !== undefined) {
+        let priceObj = storageService.getGlobal('conversionRates')
```

Comment:
I'd move this outside the if, because you're effectively doing the getGlobal twice for the same thing. Chances are performance impact is non-existent. Just thought it might be worth reducing the code smell a bit 😄

Reply:
Ye fixed up the style like ya suggested!